### PR TITLE
Temporarily remove 7.17.X from the upgrade version list until integration tests are stable again

### PR DESCRIPTION
**Pull request by pchila**
## What does this PR do?
Temporarily remove 7.17 versions from elastic-agent integration tests until tests are green again

## Why is it important?
To allow https://github.com/elastic/elastic-agent/pull/5379 to be merged in a timely manner

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

## Related issues

-

## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

File: .agent-versions.json [hunk 1]
```diff
@@ -4,8 +4,6 @@
     "8.15.0",
     "8.14.4-SNAPSHOT",
     "8.14.3",
-    "8.13.4",
-    "7.17.24-SNAPSHOT",
-    "7.17.23"
```
Comment (cmacknz) [inline .agent-versions.json:9]: Is just removing 7.17.23 enough to get past this? Or is it failing for both 7.17.x versions?

Comment (cmacknz) [reply .agent-versions.json:9]: Answer on Slack, it is failing for all of them.

**Review by cmacknz — APPROVED** [review]
Once this merges, create a PR reverting it since 8.15.2-SNAPSHOT existing should fix it and/or the 7.17 fix will fix it for 7.17.24-SNAPSHOT.

**Comment by ycombinator** [comment]
Serverless integration tests are failing on `TestLogIngestionFleetManaged/Monitoring_logs_are_shipped`.  I just reopened https://github.com/elastic/elastic-agent/issues/3741 to try and fix it again.

**Comment by ycombinator** [comment]
Integration tests are failing on `TestDebLogIngestFleetManaged/Monitoring_logs_are_shipped`.

**Comment by pchila** [comment]
> Once this merges, create a PR reverting it since 8.15.2-SNAPSHOT existing should fix it and/or the 7.17 fix will fix it for 7.17.24-SNAPSHOT.

Revert PR is https://github.com/elastic/elastic-agent/pull/5471

**Pull request merged**